sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java

```
@@ -244,13 +251,21 @@ public PlacementRule run(StringMatcher taskFilter, String fieldName, String oper
      * hostname property: {@code [["hostname", "CLUSTER", "a.specific.node.com"]]}
      */
     private static class ClusterOperator implements Operator {
-        public PlacementRule run(StringMatcher taskFilter, String fieldName, String operatorName,
-                                 Optional<String> requiredParameter) throws IOException {
+        public PlacementRule run(
+                StringMatcher taskFilter,
+                String fieldName,
+                String operatorName,
+                Optional<String> requiredParameter) throws IOException {
+
             String parameter = validateRequiredParameter(operatorName, requiredParameter);
-            if (isHostname(fieldName)) {
-                return HostnameRuleFactory.getInstance().require(ExactMatcher.create(parameter));
-            } else {
-                return AttributeRuleFactory.getInstance().require(ExactMatcher.createAttribute(fieldName, parameter));
+
+            switch (PlacementUtils.getField(fieldName)) {
+                case HOSTNAME:
+                    return HostnameRuleFactory.getInstance().require(ExactMatcher.create(parameter));
+                case ATTRIBUTE:
+                default:
```

should the default case throw an `UnsupportedOperationException`?